### ISSUE TITLE
environment variable http_proxy cannot be overridden

### ISSUE THREAD

**Issue by mybigman** [containers/libpod#6017]
/kind bug

**Description**

--env-host as no effect
http_proxy environment variable ignored.

**Steps to reproduce the issue:**

1. set http_proxy environment variable

2. ```podman run -dt --name test -e "http_proxy=http://1.1.1.1:1111" alpine:latest```

3. ```podman exec -it test sh```

4. `env`

**Describe the results you received:**

http_proxy takes on the host value regardless.
--env-host=false also has no effect

**Describe the results you expected:**

to change or unset it. 

**Additional information you deem important (e.g. issue happens only occasionally):**

**Output of `podman version`:**

```
Version:            1.8.2
RemoteAPI Version:  1
Go Version:         go1.14
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

```
debug:
  compiler: gc
  git commit: ""
  go version: go1.14
  podman version: 1.8.2
host:
  BuildahVersion: 1.14.3
  CgroupVersion: v2
  Conmon:
    package: conmon-2.0.14-1.fc32.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.14, commit: a08df20e61ed6795fd25cc9e27ef452cf3259c06'
  Distribution:
    distribution: fedora
    version: "32"
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  MemFree: 59408179200
  MemTotal: 67410444288
  OCIRuntime:
    name: crun
    package: crun-0.13-1.fc32.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.13
      commit: e79e4de4ac16da0ce48777afb72c6241de870525
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +YAJL
  SwapFree: 33772531712
  SwapTotal: 33772531712
  arch: amd64
  cpus: 24
  eventlogger: journald
  hostname: theden.localdomain
  kernel: 5.6.3-300.fc32.x86_64
  os: linux
  rootless: true
  slirp4netns:
    Executable: /usr/bin/slirp4netns
    Package: slirp4netns-1.0.0-1.fc32.x86_64
    Version: |-
      slirp4netns version 1.0.0
      commit: a3be729152a33e692cd28b52f664defbf2e7810a
      libslirp: 4.2.0
  uptime: 122h 9m 24.56s (Approximately 5.08 days)
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  ConfigFile: /home/champ/.config/containers/storage.conf
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-0.7.8-1.fc32.x86_64
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 0.7.8
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  GraphRoot: /home/champ/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 23
  RunRoot: /run/user/1000/containers
  VolumePath: /home/champ/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-1.8.2-2.fc32.x86_64
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
physical hardware

**Comment by mheon** [comment]
Hmm. Sounds like a bug.

For now, `--http-proxy=false` should disable adding proxy-related environment variables automatically.

**Comment by rhatdan** [comment]
This works in podman 1.9
```
# podman run -dt --name test -e "http_proxy=http://1.1.1.1:1111" alpine:latest
5bc8aa92c0a184cc98ce0e7f10088cabfad01a2b16b848863d5a92a9b89d94f9
# podman exec -it test env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
TERM=xterm
http_proxy=http://1.1.1.1:1111
container=podman
HOSTNAME=
HOME=/root
```

**Comment by mheon** [comment]
@rhatdan Nope, if it's set on the host, we can't override.

```
mcs@Bellerophon $ sudo su -c "export http_proxy=http://2.2.2.2:2222; podman run -dt --name test -e "http_proxy=http://1.1.1.1:1111" alpine:latest"                                                                                                    130 ↵
9a9d418c8b2881d5d089d784d0504f2919b925f232bf2010b9df4a899be32ab0
mcs@Bellerophon $ sudo podman exec -t -i 9a9d418c8b2881d5d089d784d0504f2919b925f232bf2010b9df4a899be32ab0 bash
Error: executable file not found in $PATH: No such file or directory: OCI runtime command not found error
mcs@Bellerophon $ sudo podman exec -t -i 9a9d418c8b2881d5d089d784d0504f2919b925f232bf2010b9df4a899be32ab0 sh                                                                                                                                          127 ↵
/ # printenv
HOSTNAME=
SHLVL=1
HOME=/root
container=podman
http_proxy=http://2.2.2.2:2222
TERM=xterm
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
PWD=/
/ # 

```

**Comment by mheon** [comment]
```
mcs@Bellerophon $ podman -v
podman version 1.9.0
```

**Comment by rhatdan** [comment]
Ok so this sounds like a containers.conf thing?

**Comment by mheon** [comment]
or the http_proxy flag - don't think that one is handled by containers.conf

**Comment by mheon** [comment]
Fixed in master and v1.9.2
